source/docs/software/advanced-control/state-space/state-space-flywheel-walkthrough.rst

```diff
@@ -1,4 +1,123 @@
+
+.. math:: 
+    \mathbf{\dot{x}} &= \begin{bmatrix}\frac{-kV}{kA} \end{bmatrix} \mathbf{x} + \begin{bmatrix}\frac{1}{kA} \end{bmatrix} \mathbf{u} \\
+    \mathbf{y} &= \begin{bmatrix}1\end{bmatrix} \mathbf{x} + \begin{bmatrix}0\end{bmatrix} \mathbf{u}
+
+Where :math:`\mathbf{y}` is the flywheel's velocity, as measured by a sensor of some kind.
+
+The ``LinearSystem`` class contains methods for easily creating state-space systems indentified using :term:`system identification`. This example shows a flywheel model with a kV of 1 and a kA of 0.5:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        import edu.wpi.first.wpilibj.system.LinearSystem;
+
+        var kv = 1.0;
+        var ka = 0.5;
+        var model = LinearSystem.identifyVelocitySystem(kv, ka, 12.0);
+
+    .. code-tab:: cpp
+
+        #include <frc/system/LinearSystem.h>
+        #include <frc/system/plant/LinearSystemId.h>
+
+        double kv = 1.0;
+        double ka = 0.5;
+        auto system = frc::IdentifyVelocitySystem(kv, ka, 12_V);
+
+Modeling using flywheel Moment of Intertia and gearing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A flywheel can also be modeled without access to a physical robot, using information about the motors, gearing and flywheel moment of inertia. A full derivation is presented in Section 8.2.1 of  `Controls Engineering in FRC <https://file.tavsys.net/control/controls-engineering-in-frc.pdf>`__.
+
+The ``LinearSystem`` class contains methods to easily create a model of a flywheel from the flywheel's motors, gearing and moment of inertia. The moment of inertia can be calculated using CAD software or using physics. The examples used here are detailed in the flywheel example (TODO link to allwpilib example). Note that the gearing is written as output over input -- that is, if the flywheel spins slower than the motors, this number should be greater than one.
+
+.. note:: The C++ LinearSystem class uses :ref:`docs/software/basic-programming/cpp-units:The C++ Units Library` to prevent unit mixups and assert dimensionality. 
+
+.. tabs::
+
+    .. code-tab:: java
+
+        private static final double kFlywheelMomentOfInertia = 0.00032; // kg * m^2
+        private static final double kFlywheelGearing = 1.0;
+
+        private final LinearSystem<N1, N1, N1> m_flywheelPlant = LinearSystem.createFlywheelSystem(
+                DCMotor.getNEO(2),
+                kFlywheelMomentOfInertia,
+                kFlywheelGearing,
+                12.0);
+
+    .. code-tab:: cpp
+    
```

Comment:
Add includes like the example above.

Reply:
For now these are remote literal includes